test/features/pagecontent/rerendering.js

```diff
@@ -6,46 +6,150 @@
+            uri: server.config.bucketURL + dynamic1
         })
         .then(function (res) {
             assert.deepEqual(res.status, 200);
             r1tid1 = res.headers.etag;
             hasTextContentType(res);
 
+            // delay for 1s to make sure that the timestamp differs on re-render
+            return P.delay(1500)
+            .then(function() {
+                return preq.get({
+                    uri: server.config.bucketURL + dynamic1,
+                    headers: { 'cache-control': 'no-cache' }
+                });
+            });
+        })
+        .then(function (res) {
+            // Since this is a dynamic page which should render the same each
+            // time, the tid should not change.
+            r1tid2 = res.headers.etag;
+            assert.notDeepEqual(r1tid2, r1tid1);
+            assert.notDeepEqual(r1tid2, undefined);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic1 + '/' + r1tid1
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid1);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic1
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid2);
+            hasTextContentType(res);
+
             return preq.get({
-                uri: server.config.bucketURL + '/html/Main_Page/' + r1,
+                uri: server.config.bucketURL + dynamic1 + '/' + r1tid2
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid2);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic2
+            });
+        })
+        .then(function (res) {
+            r2tid1 = res.headers.etag;
+            assert.deepEqual(res.status, 200);
+            hasTextContentType(res);
+
+            // Delay a bit to give the async save time to complete
+            return preq.get({
+                uri: server.config.bucketURL + dynamic2 + '/' + r2tid1
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r2tid1);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic1
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid2);
+            hasTextContentType(res);
+        });
+    });
+
+    it('should render & re-render independent revisions, if-unmodified-since support', function () {
+        return preq.get({
+            uri: server.config.bucketURL + dynamic1,
+            headers: {
+                'cache-control': 'no-cache',
+                'if-unmodified-since': 'Wed Dec 11 2013 16:00:00 GMT-0800 (PST)',
```

Comment:
The date should be in [RFC1123](http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.3.1), so either PST or GMT-0800 should be there, but not both.